src/test/libcephfs/ceph_pthread_self.h

```diff
@@ -0,0 +1,19 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_TEST_LIBCEPHFS_PTHREAD_SELF
+#define CEPH_TEST_LIBCEPHFS_PTHREAD_SELF
+
+#include <pthread.h>
+
+#include <type_traits>
+
+static uint64_t ceph_pthread_self() {
```

Comment:
It would be nice to have a comment block above this that explains why it's needed for posterity. Eventually, maybe gtest will handle this in a cleaner fashion and we'll be able to revert to just using pthread_self internally rather than this wrapper.

Reply:
@jtlayton 
I can still do that.
Then the question becomes, how do we find out when `gtest` has been cleaned up?